src/main/java/org/aksw/simba/lemming/metrics/single/edgemanipulation/EdgeModifier.java

```diff
@@ -27,188 +29,218 @@
-				}else if(metric.getName().equalsIgnoreCase("#nodetriangles")){
+				} else if (metric.getName().equalsIgnoreCase("#nodetriangles")) {
 					isCountingNodeTriangles = true;
 				}
-				
+
 				double metVal = metric.apply(clonedGraph);
 				String name = metric.getName();
-				LOGGER.info("Value of "+ metric.getName() + " is " + metVal);
-				//compute value for each of metrics
+				LOGGER.info("Value of " + metric.getName() + " is " + metVal);
+				// compute value for each of metrics
 				mMapMetricValues.put(name, metVal);
 			}
 		}
-		if(!isCountingNodeTriangles){
+		if (!isCountingNodeTriangles) {
 			mMapMetricValues.put("#nodetriangles", 0);
 		}
-		if(!isCoutingEdgeTriangles){
+		if (!isCoutingEdgeTriangles) {
 			mMapMetricValues.put("#edgetriangles", 0);
 		}
 
-		//create a backup map metric values
+		// create a backup map metric values
 		mMapOrignalMetricValues = mMapMetricValues.clone();
 	}
-	
-	public ColouredGraph getGraph(){
+
+	public ColouredGraph getGraph() {
 		return mEdgeModification.getGraph();
 	}
-	
-	public ObjectDoubleOpenHashMap<String> tryToRemoveAnEdge(TripleBaseSingleID triple){
-		if(triple != null && triple.edgeId != -1 &&
-				triple.edgeColour != null &&
-				triple.tailId != -1 && triple.headId !=-1){
-			
-			//add to list of removed edges
-			//mLstRemovedEdges.add(triple);
-			
+
+	public ObjectDoubleOpenHashMap<String> tryToRemoveAnEdge(TripleBaseSingleID triple) {
+		if (triple != null && triple.edgeId != -1 && triple.edgeColour != null && triple.tailId != -1
+				&& triple.headId != -1) {
+
+			// add to list of removed edges
+			// mLstRemovedEdges.add(triple);
+
 			ObjectDoubleOpenHashMap<String> mapChangedMetricValues = new ObjectDoubleOpenHashMap<String>();
-			
+
 			mEdgeModification.removeEdgeFromGraph(triple.edgeId);
-			if(isCountingNodeTriangles){
+			if (isCountingNodeTriangles) {
 				int newNodeTri = mEdgeModification.getNewNodeTriangles();
-				mapChangedMetricValues.put("#nodetriangles", newNodeTri);	
+				mapChangedMetricValues.put("#nodetriangles", newNodeTri);
+			}
+
+			if (isCoutingEdgeTriangles) {
+				int newEdgeTri = mEdgeModification.getNewEdgeTriangles();
+				mapChangedMetricValues.put("#edgetriangles", newEdgeTri);
 			}
-			
-			if(isCoutingEdgeTriangles){
-		        int newEdgeTri = mEdgeModification.getNewEdgeTriangles();
-		        mapChangedMetricValues.put("#edgetriangles", newEdgeTri);				
+
+			ColouredGraph graph = mEdgeModification.getGraph();
+
+			for (SingleValueMetric metric : mLstMetrics) {
+				if (!metric.getName().equalsIgnoreCase("#edgetriangles")
+						&& !metric.getName().equalsIgnoreCase("#nodetriangles")) {
+					// double metVal = metric.apply(graph);
+					// Calling update method to get the metric values based on previous results
+					mMapPrevMetricsResultRemoveEdge.put(metric.getName(), metric.update(triple, graph, false,
+							new HashMap<>(mMapPrevMetricsResult).get(metric.getName()), mEdgeModification.getmVertexDegrees()));
```

Comment:
Why is there a new hash map created? This is very costly and not necessary!

Reply:
I have removed this logic and using the values stored in the mMapPrevMetricsResult map.